project/src/main/java/com/example/project/model/Transaction.java

```diff
@@ -0,0 +1,44 @@
+package com.example.project.model;
+
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * @Author Fusheng Tan
+ * @Version 1.0
+ */
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "transactions")
+public class Transaction {
+    @Id
+    @Column(name = "transaction_id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "time", nullable = false, updatable = false, insertable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
+    public Timestamp time;
+
+    @Column(name = "amount")
+    private BigDecimal amount;
+
+    @Column(name = "fromAccount")
+    private Long fromAccount;
+
+    @Column(name = "toAccount")
+    private Long toAccount;
```

Comment:
Hi bro you need more sleep!

Reply:
For this one, it all depends on how you design it, actually, I don't really like creating too many FKs between tables coz it will affect performance, especially for the table with massive data(The transaction table will be the one) and all these constraint stuff by FKs can be handled by good implementation logic. For example, make sure the account is existing in our DB before allowing the user to do the transaction

Reply:
And for the toAccount attribute, are we allowing the user to transfer money from our app to accounts by other banks? If so, we should not map to our account table as well coz we don't have all the possible accounts.